src/docdb/nodes.ts

```diff
@@ -119,6 +119,9 @@ export class DocDBCollectionNode implements INode {
 		this._children = this._children.concat(loadMoreDocuments.map(document => new DocDBDocumentNode(document.id, this, document)));
 	}
 
+	addNewDocToCache(document: any): void {
```

Comment:
`addNewDocToCache` is just a line and used just once for each definition (in docdb and mongo). Would it make sense to just have the unshift operation just be inline?

Reply:
@PrashanthCorp I would prefer to keep this._children as private. That's the only reason I created this one-line public method